extra_geom/tests/test_pnccd_geometry.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+from matplotlib.axes import Axes
+
+from extra_geom import PNCCDGeometry
+
+def test_positions():
+    # Positional constants
+    gap_pixels = 50
+    offset_pixels = 10
+
+    # Calculate parameters for the relative/absolute methods
+    top_offset = (-offset_pixels * PNCCDGeometry.pixel_size, 0, 0)
+    bottom_offset = (offset_pixels * PNCCDGeometry.pixel_size, 0, 0)
+    absolute_pos = np.array(
+        [(-PNCCDGeometry.frag_fs_pixels // 2 - offset_pixels,
+          PNCCDGeometry.frag_ss_pixels + gap_pixels // 2,
+          0),
+         (-PNCCDGeometry.frag_fs_pixels // 2 + offset_pixels,
+          -gap_pixels // 2,
+          0)]
+    ) * PNCCDGeometry.pixel_size
+
+    # Create the geometries
+    absolute_geom = PNCCDGeometry.from_absolute_positions(top=absolute_pos[0], bottom=absolute_pos[1])
+    relative_geom = PNCCDGeometry.from_relative_positions(gap=gap_pixels * PNCCDGeometry.pixel_size,
+                                                          top_offset=top_offset,
+                                                          bottom_offset=bottom_offset)
+
+    # They should be the same
+    assert all(np.array_equal(abs_geom_mod[0].centre(), rel_geom_mod[0].centre())
+               for abs_geom_mod, rel_geom_mod
+               in zip(absolute_geom.modules, relative_geom.modules))
+
+def test_snap_assemble_data():
+    gap_pixels = 100
+    geom = PNCCDGeometry.from_relative_positions(gap=gap_pixels * PNCCDGeometry.pixel_size)
+
+    def check_result(shape):
+        stacked_data = np.zeros(shape)
+        img, centre = geom.position_modules_fast(stacked_data)
+        expected_img_shape = (PNCCDGeometry.frag_ss_pixels * PNCCDGeometry.n_modules + gap_pixels,
+                              PNCCDGeometry.frag_fs_pixels)
+
+        assert img.shape == expected_img_shape
+        assert tuple(centre) == (expected_img_shape[0] // 2, expected_img_shape[1] // 2)
+        # Everywhere in the gap should be NaN
+        assert np.isnan(img[img.shape[0] // 2, 0])
+        assert img[50, 50] == 0
+
+    # No re-shaping necessary
+    check_result(PNCCDGeometry.expected_data_shape)
+    # The class should automatically reshape the data
+    check_result((PNCCDGeometry.frag_ss_pixels * PNCCDGeometry.n_modules, PNCCDGeometry.frag_fs_pixels))
```

Comment:
You mind parametrizing `shape` in this test using `@pytest.mark.parametrize`? It might make the intent a bit more clearer than with nested function.

Reply:
Ah yeah, that's much nicer. Fixed in 95b193a.